src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/comments/ComposerWithCommentEnabledTest.kt

```diff
@@ -0,0 +1,488 @@
+
+        assertNodesEqual(expected, result)
+    }
+
+    test("all comments 3") {
+        val data = "" + //
+        "# Block Comment1\n" + //
+            "[ item1, item2: value2, {item3: value3} ] # Inline Comment1\n" + //
+            "# Block Comment2\n" + //
+            ""
+        val expected = listOf(
+            "Block Comment", //
+            "SequenceNode", //
+            "    ScalarNode: item1", //
+            "    MappingNode", //
+            "        Tuple", //
+            "            ScalarNode: item2", //
+            "            ScalarNode: value2", //
+            "    MappingNode", //
+            "        Tuple", //
+            "            ScalarNode: item3", //
+            "            ScalarNode: value3", //
+            "    InLine Comment", //
+            "End Comment", //
+        )
+
+        val sut = newComposerWithCommentsEnabled(data)
+        val result = getNodeList(sut)
+
+        assertNodesEqual(expected, result)
+    }
+
+    test("get single node") {
+        val data = "" + //
+            "\n" + //
+            "abc: def # commment\n" + //
+            "\n" + //
+            "\n"
+        val expected = listOf(
+            "MappingNode", //
+            "    Tuple", //
+            "        Block Comment", "        ScalarNode: abc", //
+            "        ScalarNode: def", //
+            "            InLine Comment", //
+            "End Comment", //
+            "End Comment", //
+        )
+
+        val sut = newComposerWithCommentsEnabled(data)
+        val result = getNodeList(sut)
+
+        assertNodesEqual(expected, result)
+    }
+
+    test("get single node header comment") {
+        val data = "" + //
+            "\n" + //
+            "# Block Comment1\n" + //
+            "# Block Comment2\n" + //
+            "abc: def # commment\n" + //
+            "\n" + //
+            "\n"
+        val expected = listOf(
+            "MappingNode", //
+            "    Tuple", //
+            "        Block Comment", //
+            "        Block Comment", //
+            "        Block Comment", //
+            "        ScalarNode: abc", //
+            "        ScalarNode: def", //
+            "            InLine Comment", //
+            "End Comment", //
+            "End Comment", //
+        )
+
+        val sut = newComposerWithCommentsEnabled(data)
+        val result = getNodeList(sut)
+
+        assertNodesEqual(expected, result)
+    }
+
+    test("base constructor get data") {
+        val data = "" + //
+            "\n" + //
+            "abc: def # comment\n" + //
+            "\n" + //
+            "\n"
+        val sut = TestConstructor(LoadSettings.builder().build())
+        val composer = newComposerWithCommentsEnabled(data)
+        val result = sut.constructSingleDocument(composer.getSingleNode())
+        (result is LinkedHashMap<*, *>) shouldBe true
+        @Suppress("UNCHECKED_CAST")
+        val map = result as LinkedHashMap<String, Any>
```

Comment:
cast can be replaced with an existing assertion from kotest library
```suggestion
        val map = result.shouldBeInstanceOf<LinkedHashMap<String, Any>>()
```